### DISCUSSION TITLE
how to filter relation by role?

### DISCUSSION THREAD

**Discussion by andreskrip** [spatie/laravel-permission#2028] [Q&A]
Hi, is it possible that only the user who has the role can get relation data, and the rest of the users get an empty relation?

I tried the https://github.com/spatie/laravel-permission/issues/992#issuecomment-452046771 solution:

```
public function connections(): BelongsToMany
    {
        return $this->belongsToMany(Connection::class, 'connection_users', 'user_id', 'connection', 'id', 'connection')->role('operator');
    }
```
but it doesn't work (BadMethodCallException. Call to undefined method Illuminate\Database\Eloquent\Relations\BelongsToMany::role())

**Comment by erikn69** [comment]
Maybe with join to `model_has_roles` and join with `roles`, then you can use `->where('roles.name','operator')`
